finagle-mysql/src/main/scala/com/twitter/finagle/mysql/CanBeParameter.scala

```diff
@@ -0,0 +1,182 @@
+      def typeCode(param: Float) = Type.Float
+      def write(writer: BufferWriter, param: Float) = writer.writeFloat(param)
+    }
+  }
+
+  implicit val doubleCanBeParameter = {
+    new CanBeParameter[Double] {
+      def sizeOf(param: Double) = 8
+      def typeCode(param: Double) = Type.Double
+      def write(writer: BufferWriter, param: Double) = writer.writeDouble(param)
+    }
+  }
+
+  implicit val byteArrayCanBeParameter = {
+    new CanBeParameter[Array[Byte]] {
+      def sizeOf(param: Array[Byte]) = Buffer.sizeOfLen(param.length) + param.length
+      def typeCode(param: Array[Byte]) = {
+        if (param.length <= 255) Type.TinyBlob
+        else if (param.length <= 65535) Type.Blob
+        else if (param.length <= 16777215) Type.MediumBlob
+        else -1
+      }
+      def write(writer: BufferWriter, param: Array[Byte]) = writer.writeLengthCodedBytes(param)
+    }
+  }
+
+  implicit val valueCanBeParameter = {
+    new CanBeParameter[Value] {
+      def sizeOf(param: Value) = param match {
+        case RawValue(_, _, true, b) => Buffer.sizeOfLen(b.length) + b.length
+        case StringValue(s)          => val bytes = s.getBytes(Charset.defaultCharset); Buffer.sizeOfLen(bytes.size) + bytes.size
+        case ByteValue(_)            => 1
+        case ShortValue(_)           => 2
+        case IntValue(_)             => 4
+        case LongValue(_)            => 8
+        case FloatValue(_)           => 4
+        case DoubleValue(_)          => 8
+        case NullValue               => 0
+        case _                       => 0
+      }
+
+      def typeCode(param: Value) = param match {
+        case RawValue(typ, _, _, _) => typ
+        case StringValue(_)         => Type.VarChar
+        case ByteValue(_)           => Type.Tiny
+        case ShortValue(_)          => Type.Short
+        case IntValue(_)            => Type.Long
+        case LongValue(_)           => Type.LongLong
+        case FloatValue(_)          => Type.Float
+        case DoubleValue(_)         => Type.Double
+        case EmptyValue             => -1
+        case NullValue              => Type.Null
+      }
+
+      def write(writer: BufferWriter, param: Value) = param match {
+        // allows for generic binary values as params to a prepared statement.
+        case RawValue(_, _, true, bytes) => writer.writeLengthCodedBytes(bytes)
+        // allows for Value types as params to prepared statements
+        case ByteValue(b)                => writer.writeByte(b)
+        case ShortValue(s)               => writer.writeShort(s)
+        case IntValue(i)                 => writer.writeInt(i)
+        case LongValue(l)                => writer.writeLong(l)
+        case FloatValue(f)               => writer.writeFloat(f)
+        case DoubleValue(d)              => writer.writeDouble(d)
+        case StringValue(s)              => writer.writeLengthCodedString(s)
+        case _                           => ()
```

Comment:
@mosesn I have tried to mirror the existing behavior as faithfully as possible. I think other values are skipped in current implementation too. As for `null`s, we are not handling them here. Should they be handled here? Handling `null`s everywhere on a case-by-case basis seems messy, and something to be avoided.